source/d3d12/d3d12_device.cpp

```diff
@@ -513,7 +513,9 @@ D3D12_HEAP_PROPERTIES STDMETHODCALLTYPE D3D12Device::GetCustomHeapProperties(UIN
 }
 HRESULT STDMETHODCALLTYPE D3D12Device::CreateCommittedResource(const D3D12_HEAP_PROPERTIES *pHeapProperties, D3D12_HEAP_FLAGS HeapFlags, const D3D12_RESOURCE_DESC *pResourceDesc, D3D12_RESOURCE_STATES InitialResourceState, const D3D12_CLEAR_VALUE *pOptimizedClearValue, REFIID riidResource, void **ppvResource)
 {
-	return _orig->CreateCommittedResource(pHeapProperties, HeapFlags, pResourceDesc, InitialResourceState, pOptimizedClearValue, riidResource, ppvResource);
+	D3D12_RESOURCE_DESC new_desc = *pResourceDesc;
+	new_desc.Flags = new_desc.Flags & ~D3D12_RESOURCE_FLAG_DENY_SHADER_RESOURCE;
```

Comment:
We should probably check wether the resource format is a depth format and only remove the flag for those. There might be a performance implication, so it's better to avoid removing the flag for textures that are not interesting to ReShade anyway.

Reply:
That's what i thinked about, but it seems that this flag is only allowed for depthstencil resources, so a check may be not necessary...

Reply:
Ah, right, just read that in the documentation. Ok then.